admiral/pkg/clusters/serviceentry.go

```diff
@@ -179,6 +196,26 @@ func modifyServiceEntryForNewServiceOrPod(event admiral.EventType, env string, s
 	return serviceEntries
 }
 
+func updateEndpointsForBlueGreen(rollout *argo.Rollout, weightedServices map[string]*WeightedService, cnames map[string]string,
+	ep *networking.ServiceEntry_Endpoint, sourceCluster string, meshHost string) {
+	activeServiceName := rollout.Spec.Strategy.BlueGreen.ActiveService
+	previewServiceName := rollout.Spec.Strategy.BlueGreen.PreviewService
+
+	if previewService, ok := weightedServices[previewServiceName]; strings.HasPrefix(meshHost, common.BlueGreenRolloutPreviewPrefix+common.Sep) && ok {
+		previewServiceInstance := previewService.Service
+		localFqdn := previewServiceInstance.Name + common.Sep + previewServiceInstance.Namespace + common.DotLocalDomainSuffix
+		cnames[localFqdn] = "1"
+		ep.Address = localFqdn
+		ep.Ports = GetMeshPortsForRollout(sourceCluster, previewServiceInstance, rollout)
+	} else if activeService, ok := weightedServices[activeServiceName]; ok {
```

Comment:
Shouldn't this be `if` instead of `else if`, will this method (updateEndpointsForBlueGreen) be called twice once for active and preview endpoints?

Reply:
Yes, this will be called twice, once for active and preview endpoints.

Reply:
If `meshHost` is a preview host then `else if` should not execute. Making it as `if` will overwrite with active service address to preview endpoint.